core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/GroupByAggregator.java

```diff
@@ -393,4 +411,64 @@ private Schema getGroupKeySchema(Schema inputSchema) {
     }
     return Schema.recordOf("group.key.schema", fields);
   }
+
+  @Override
+  public Relation transform(RelationalTranformContext relationalTranformContext, Relation relation) {
+    Optional<ExpressionFactory<String>> expressionFactory = relationalTranformContext
+      .getEngine().getExpressionFactory(StringExpressionFactoryType.SQL);
+    if (!expressionFactory.isPresent()) {
+      return new InvalidRelation("Cannot find an Expression Factory");
+    }
+
+    GroupByAggregationDefinition aggregationDefinition = getAggregationDefinition(expressionFactory.get(), relation);
+
+    if (aggregationDefinition == null) {
+      return new InvalidRelation("Cannot create a relation as aggregation definition is invalid");
+    }
+    return relation.groupBy(aggregationDefinition);
+  }
+
+  private GroupByAggregationDefinition getAggregationDefinition(ExpressionFactory<String> expressionFactory,
+                                                                Relation relation) {
+    List<Expression> groupByExpressions = new ArrayList<>(groupByFields.size());
+    Map<String, Expression> selectExpressions = new HashMap<>();
+
+    for (String field: groupByFields) {
+      String columnName = getColumnName(expressionFactory, relation, field);
+      Expression groupByExpression = expressionFactory.compile(columnName);
+      groupByExpressions.add(groupByExpression);
+      selectExpressions.put(field, groupByExpression);
+    }
+
+    for (GroupByConfig.FunctionInfo aggregate: functionInfos) {
+      String alias = aggregate.getName();
+      String columnName = getColumnName(expressionFactory, relation, aggregate.getField());
+      GroupByConfig.Function function = aggregate.getFunction();
```

Comment:
I've called this out in the PR description - should the decision on whether the function is supported or not be taken in the engine rather than in the aggregate definition since the list of supported functions can be different depending on engine?